bitcoin/src/pow.rs

```diff
@@ -241,7 +242,10 @@ impl Target {
     ///
     /// [`difficulty`]: Target::difficulty
     #[cfg_attr(all(test, mutate), mutate)]
-    pub fn difficulty_float(&self) -> f64 { TARGET_MAX_F64 / self.0.to_f64() }
+    pub fn difficulty_float(&self) -> f64 {
+        debug_assert_ne!(self.0, U256::ZERO);
+        TARGET_MAX_F64 / self.0.to_f64()
```

Comment:
In 14ab447e37411cde144eb81bad10ba5b425dcf3f:

In the f64 case we'll get +- `Inf` if we divide by zero. I wonder if we should just preserve that behavior rather than having a debug assertion.

Reply:
In 7163eb91c5cd6d5aa66ededdcb95c8cd9f9eccf3 I mean.

Reply:
Good observation. TIL div by zero returns `f64::NAN`. I've removed the assertion and documented the return value.